opennms-container/minion/Dockerfile

```diff
@@ -1,88 +1,64 @@
+# To avoid issues, we rearrange the directories in pre-stage to avoid injecting these
+# as addtional layers into the final image.
 ##
-ARG BASE_IMAGE="opennms/openjdk"
-ARG BASE_IMAGE_VERSION="11.0.5.10-b3108"
+ARG BASE_IMAGE="opennms/deploy-base:jre-1.0.0.b31"
 
-FROM ${BASE_IMAGE}:${BASE_IMAGE_VERSION} as minion-base
+FROM ${BASE_IMAGE} as minion-base
 
-ARG REQUIRED_RPMS="wget gettext jicmp jicmp6 openssh-clients"
+ADD ./tarball/minion.tar.gz /opt/
 
-ARG REPO_KEY_URL="https://yum.opennms.org/OPENNMS-GPG-KEY"
-ARG REPO_RPM="https://yum.opennms.org/repofiles/opennms-repo-stable-rhel8.noarch.rpm"
+# Organize files based on the build home dir /opt/minion
+RUN mv /opt/minion-* /opt/minion
 
-SHELL ["/bin/bash", "-c"]
-
-# Collect generic steps in a layer for caching
-RUN rpm --import "${REPO_KEY_URL}" && \
-    dnf -y install epel-release && \
-    dnf -y install "${REPO_RPM}" && \
-    dnf -y install ${REQUIRED_RPMS} && \
-    dnf clean all && \
-    rm -rf /var/cache/yum
-
-# Allow to send ICMP messages as non-root user
-RUN setcap cap_net_raw+ep ${JAVA_HOME}/bin/java && \
-    echo ${JAVA_HOME}/lib/jli > /etc/ld.so.conf.d/java-latest.conf && \
-    ldconfig
-
-# Create Minion user with a specific group ID
-RUN groupadd -g 10001 minion && \
-    adduser -u 10001 -g 10001 -d /opt/minion -s /usr/bin/bash minion
-
-# Create SSH Key-Pair to use with the Karaf Shell
-RUN mkdir /opt/minion/.ssh && \
-    chmod 700 /opt/minion/.ssh && \
-    ssh-keygen -t rsa -f /opt/minion/.ssh/id_rsa -q -N ""
+# We set the correct permissions in this stage, so we don't have this as an addtional
+# layer in our deploy image.
+RUN groupadd --gid 10001 minion && \
+    useradd --system --uid 10001 --gid minion minion --home-dir /opt/minion && \
+    chown 10001 /opt/minion -R && \
+    chgrp -R 0 /opt/minion && \
+    chmod -R g=u /opt/minion
 
 ##
-# Install and setup OpenNMS Minion RPMS
+# Prod image with minimal image size
 ##
-FROM minion-base
+FROM ${BASE_IMAGE}
 
-ARG USER="minion"
+# Create user to be able to run as non root with a known user and group id.
+RUN groupadd --gid 10001 minion && \
+    useradd --system --uid 10001 --gid minion minion --home-dir /opt/minion
 
-# A default which installs the minimum required Minion packages
-ARG MINION_PACKAGES="opennms-minion"
+# Install entrypoint wrapper and health check script
+COPY container-fs/entrypoint.sh /
+COPY container-fs/health.sh /
 
-# Allow to install optional packages via YUM
-ARG ADD_YUM_PACKAGES
+# Install confd configuration
+COPY ./container-fs/confd/ /opt/minion/confd/
 
-COPY ./rpms /tmp/rpms
+# If you copy from /opt/minion to /opt/minion the permissions are not preserved
+# We would have 755 for minion:root instead of 775 and prevents writing lock files in /opt/minion
+COPY --chown=10001:0 --from=minion-base /opt /opt
 
-SHELL ["/bin/bash", "-c"]
+# Install confd.io configuration files and scripts and ensure they are executable
+COPY ./container-fs/confd/ /opt/minion/confd/
```

Comment:
Do we need line `45` given line `38`?

Reply:
Thank you! Very good catch slipped through a resolved merge conflict.